Framework/Algorithms/test/CalculateIqtTest.h

```diff
@@ -0,0 +1,96 @@
+#ifndef CALCULATEIQTTEST_H_
+#define CALCULATEIQTTEST_H_
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAlgorithms/CalculateIqt.h"
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+
+class CalculateIqtTest : public CxxTest::TestSuite {
+public:
+  static CalculateIqtTest *createSuite() { return new CalculateIqtTest(); }
+  static void destroySuite(CalculateIqtTest *suite) { delete suite; }
+  MatrixWorkspace_sptr m_sampleWorkspace;
+  MatrixWorkspace_sptr m_resolutionWorkspace;
+
+  CalculateIqtTest() {
+    m_sampleWorkspace = setUpWorkspace(true);
+    m_resolutionWorkspace = setUpWorkspace(false);
+  }
+
+  void test_execSuccess() {
+    auto algorithm =
+        calculateIqtAlgorithm(m_sampleWorkspace, m_resolutionWorkspace);
+    TS_ASSERT_THROWS_NOTHING(algorithm->execute());
+    TS_ASSERT(algorithm->isExecuted());
+    MatrixWorkspace_sptr outWorkspace =
+        algorithm->getProperty("OutputWorkspace");
+    TS_ASSERT_EQUALS(outWorkspace->getNumberHistograms(), 1);
+    TS_ASSERT_EQUALS(outWorkspace->blocksize(), 6);
+  }
+
+  void test_invalidEnergyBounds() {
```

Comment:
Consider using propositional test names, i.e. in this case "test_throwsIfEnergyBoundsInvalid".